cmd/dosa/scopemd.go

```diff
@@ -0,0 +1,127 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/uber-go/dosa"
+)
+
+// ScopeList contains data for executing scope truncate command.
+type ScopeList struct {
+	*ScopeCmd
+}
+
+func newScopeList(provideClient mdClientProvider) *ScopeList {
+	return &ScopeList{
+		ScopeCmd: &ScopeCmd{
+			provideMDClient: provideClient,
+		},
+	}
+}
+
+// Execute executes a scope list command
+func (c *ScopeList) Execute(args []string) error {
+	client, err := c.makeClient()
+	if err != nil {
+		return errors.Wrap(err, "could not make client")
+	}
+	defer shutdownMDClient(client)
+
+	var md dosa.ScopeMetadata
+	scanOp := dosa.NewScanOp(&md).Limit(100)
+	for {
+		ctx, cancel := context.WithTimeout(context.Background(), options.Timeout.Duration())
+		defer cancel()
+
+		scopes, token, err := client.ScanEverything(ctx, scanOp)
+		if err != nil {
+			if dosa.ErrorIsNotFound(err) {
+				err = nil
+				break
+			}
+			fmt.Printf("MD table scan failed (token=%q): %v\n", token, err)
+			continue
+		}
+		scanOp.Offset(token)
+
+		for _, e := range scopes {
+			md := e.(*dosa.ScopeMetadata)
+			fmt.Printf("%+q\n", md)
+		}
+	}
+	return err
+}
+
+// ScopeShow displays metadata for the specified scopes.
+type ScopeShow struct {
+	*ScopeCmd
+	Args struct {
+		Scopes []string `positional-arg-name:"scopes" required:"1"`
+	} `positional-args:"yes" required:"1"`
+}
+
+func newScopeShow(provideClient mdClientProvider) *ScopeShow {
+	return &ScopeShow{
+		ScopeCmd: &ScopeCmd{
+			provideMDClient: provideClient,
+		},
+	}
+}
+
+// Execute shows MD for the scope(s)
+func (c *ScopeShow) Execute(args []string) error {
+	client, err := c.makeClient()
+	if err != nil {
+		return errors.Wrap(err, "could not make client")
+	}
+	defer shutdownMDClient(client)
+
+	for _, scope := range args {
+		if md, err := c.getMetadata(client, scope); err != nil {
+			fmt.Printf("Could not read scope metadata for %q: %v\n", scope, err)
+		} else {
+			fmt.Printf("%+v\n", md)
+		}
+	}
+	return nil
+}
+
+func (c *ScopeShow) getMetadata(client dosa.Client, scope string) (*dosa.ScopeMetadata, error) {
```

Comment:
Here it only takes scope as input without prefix. 
In dev, one scope has one name prefix
in prod, one scope has many prefixes. How does this method support that?

Reply:
Currently, this only handles scopes, i.e. dev. Prefix support needs to be added, there is this comment in the defn for `ScopeMetadata`:

```
The scope may be qualified by a prefix, as in "production.vsoffers".
```

Reply:
Add a comment on top of the method